designs/2025-rule-tester-assertion-options/README.md

```diff
@@ -0,0 +1,391 @@
+@@ -1239,22 +1265,22 @@ class RuleTester {
+                                                const actualLocation = {};
+                                                const expectedLocation = {};
+
+-                                               if (hasOwnProperty(error, "line")) {
++                                               if (hasOwnProperty(error, "line") || requireLocation) {
+                                                        actualLocation.line = message.line;
+                                                        expectedLocation.line = error.line;
+                                                }
+
+-                                               if (hasOwnProperty(error, "column")) {
++                                               if (hasOwnProperty(error, "column") || requireLocation) {
+                                                        actualLocation.column = message.column;
+                                                        expectedLocation.column = error.column;
+                                                }
+
+-                                               if (hasOwnProperty(error, "endLine")) {
++                                               if (hasOwnProperty(error, "endLine") || requireLocation) {
+                                                        actualLocation.endLine = message.endLine;
+                                                        expectedLocation.endLine = error.endLine;
+                                                }
+
+-                                               if (hasOwnProperty(error, "endColumn")) {
++                                               if (hasOwnProperty(error, "endColumn") || requireLocation) {
+                                                        actualLocation.endColumn = message.endColumn;
+                                                        expectedLocation.endColumn = error.endColumn;
+                                                }
+````
+
+## Documentation
+
+This RFC will be documented in the RuleTester documentation, explaining the new options and how to use them.
+So mainly here: https://eslint.org/docs/latest/integrate/nodejs-api#ruletester
+Additionally, we should write a short blog post to announce for plugin maintainers, to raise awareness of the new options and encourage them to use them in their tests.
+
+## Drawbacks
+
+This proposal adds slightly more complexity to the RuleTester logic, as it needs to handle the new options and enforce the assertions based on them.
+Currently, the RuleTester logic is already deeply nested, so adding more options may make it harder to read and maintain.
+
+Due to limitations of the `RuleTester`, the error message does not point to the correct location of the test case.
+While this is true already, it gets slightly more relevant, since now it might report missing properties, that cannot be searched for in the test file.
```

Comment:
I'm not sure if if gets more relevant, because the failing test case prints `code` or `name`, the same as now when it fails for any reason, and it can be searched by those values.

https://github.com/eslint/eslint/blob/f46fc6c137c951bc73cf3bd9446053c1b11f769b/lib/rule-tester/rule-tester.js#L1556-L1557

Reply:
Most tests I've seen so far dont have a name and the code is abiguous at best especially of the same code is run with different options.
In most cases, I actually had trouble locating the failing test in the test sets.

Reply:
> Most tests I've seen so far dont have a name and the code is abiguous at best especially of the same code is run with different options.
> In most cases, I actually had trouble locating the failing test in the test sets.

I agree, just don't see why this feature makes the mentioned limitations more relevant than they are now.

Reply:
You cannot search for something that isnt there.

But I can remove it, if you want.